testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/extension/CamundaProcessTestContextImpl.java

```diff
@@ -138,6 +140,25 @@ public JobWorkerMock mockJobWorker(final String jobType) {
     return new JobWorkerMockImpl(jobType, client);
   }
 
+  @Override
+  public void mockChildProcess(final String childProcessId, final Map<String, Object> variables) {
+    final CamundaClient client = createClient();
+    final String jobType = "mocked-child-job";
+    final BpmnModelInstance processModel =
+        Bpmn.createExecutableProcess(childProcessId)
+            .startEvent()
+            .serviceTask("child-service-task")
+            .zeebeJobType(jobType)
+            .endEvent("child-end")
+            .done();
+    client
+        .newDeployResourceCommand()
+        .addProcessModel(processModel, "test-process.bpmn")
+        .send()
+        .join();
+    mockJobWorker(jobType).thenComplete(variables);
```

Comment:
:wrench: We could set the variables directly on the end event using output mappings instead of using a service task.

We could use the client's JSON mapper to transform the variable values.